lib/rust/ensogl/core/src/display/scene.rs

```diff
@@ -937,6 +950,8 @@ impl SceneData {
         // setups now, so we are using here the main camera only.
         let camera = self.camera();
         let fullscreen_vis_camera = self.layers.panel.camera();
+        // for now, we use the same camera as `fullscreen_vis`, it might change in the future
```

Comment:
1. Comments should be sentences – starting with upper case letter and finishing with dot (please fix the comment 2 lines above too).
2. What does it mean "for now"? How someone reading this code now, in one month, in 6 months and in a year should understand it? If you are making such a decision, it should be clear why it was done. When reading this comment I do not understand the reasoning for it. If you want to use the same camera, please justify why this is the best decision (btw, I think this is a bad decision, because I do not understand why camera of project chooser should be the same as camera of full-screen visualizations).
3. ⚠️ This is a broken implementation. You are not really using the same camera. Your `layers.panel` still have a distinct camera, you are just overriding its values with `fullscreen_vis_camera` whenever the later changes. So developers still can for example try to move this camera, like `layers.panel.camera.mod_position(|p| p.x += 10.0)` and nothing will happen, which is a corrupted implementation. If you like to use the same camera, the camera should be cloned and both layers should reference the same camera instead.

Reply:
> 3. warning This is a broken implementation. You are not really using the same camera. Your `layers.panel` still have a distinct camera, you are just overriding its values with `fullscreen_vis_camera` whenever the later changes. So developers still can for example try to move this camera, like `layers.panel.camera.mod_position(|p| p.x += 10.0)` and nothing will happen, which is a corrupted implementation. If you like to use the same camera, the camera should be cloned and both layers should reference the same camera instead.

Actually, I don't see the danger you describe. First, please note that there is a distinction between `self.layers.panel` and `self.dom.layers.panel` - the first is our normal panel with EnsoGL-rendered components, the other is the DOM layer created in this PR. The first has the `Camera` structure, the second has the `view_projection_dom` as a div element.

And now:
1. Someone moves the `self.layers.panel` camera
2. It causes `update_camera`
3. both `self.dom.fullscreen_vis` and `self.dom.panel` are updated with the new state of `self.layers.panel.camera()`
4. We see elements of both layer moved.

Reply:
Oh, good catch, thanks @farmaazon for clarification. Ok, so let's fix points 1 and 2 now. Also, it would be good to add comment there describing what Adam described above. It is a tricky thing to understand without getting deeply into the codebase.

Reply:
The comment was fixed. I think it is much cleaner now because of a different naming. I struggle to write a good explanation why we *actually* use the same camera, so I kept it minimal. If someone would come up with a good one-liner, I add it.